rpcs3/Emu/System.cpp

```diff
@@ -1647,9 +1647,36 @@ game_boot_result Emulator::Load(const std::string& title_id, bool add_only, bool
 			{
 				if (!fs::is_file(g_cfg.vfs.get_dev_flash() + "sys/external/liblv2.sprx"))
 				{
-					if (!GetCallbacks().on_missing_fw())
+					const auto libs = g_cfg.core.libraries_control.get_set();
+
+					extern const std::map<std::string_view, int> g_prx_list;
+
+					// Check if there are any firmware SPRX which may be LLEd during emulation 
+					bool needs_fw = false;
+
+					for (const auto& lib : g_prx_list)
+					{
+						if (libs.count(std::string(lib.first) + ":lle") || (!lib.second && !libs.count(std::string(lib.first) + ":hle")))
+						{
+							needs_fw = true;
+							break;
+						}
+					}
+
+					// Don't prompt GUI confirmation if there aren't any
+					if (needs_fw)
 					{
 						Stop();
+
+						CallAfter([this]()
+						{
+							// Sanity check because we are using GUI signals for it, could have been installed already
+							if (!fs::is_file(g_cfg.vfs.get_dev_flash() + "sys/external/liblv2.sprx"))
```

Comment:
how is anyone supposed to install the firmware in the time it takes to process the last 20 lines?

Reply:
Fixed.